src/include/souffle/SouffleInterface.h

```diff
@@ -41,6 +41,10 @@ class tuple;
  * Object-oriented wrapper class for Souffle's templatized relations.
  */
 class Relation {
+public:
+    // you have bigger fish to fry if you have > 255 attributes
+    using arity_type = uint8_t;
```

Comment:
Is there a reason to limit it uint8_t? Could we have uint32_t instead?

Reply:
We could, but is there a reason to? It was meant more as a cheap sancheck/type-assertion that we haven't considered relations with absurdly massive tuple.

Reply:
I can think of examples (transposes of relations) where more than 256 attributes may be possible.
Hence, I don't want to restrict arities to 8bit, artificially.

Reply:
I have no idea how they live with the compile times for the data structures.